src/vsi.rs

```diff
@@ -7,6 +7,39 @@ use gdal_sys::{VSIFCloseL, VSIFileFromMemBuffer, VSIFree, VSIGetMemFileBuffer, V
 use crate::errors::{GdalError, Result};
 use crate::utils::{_last_null_pointer_err, _path_to_c_string};
 
+/// Read the file names from a virtual file system with optional recursion.
+pub fn read_dir<P: AsRef<Path>>(path: P, recursive: bool) -> Result<Vec<String>> {
```

Comment:
Just as an open question. Do we want to return a list of `String`s or of `PathBuf`s?

Reply:
I originally had it as PathBuf but changed it because it's easy to go from String to Path but not so easy going from PathBuf to &str. I can change it if you want.

Reply:
I think `PathBuf` is indeed more idiomatic here. Do we take paths as `&str` in other places?